File: PythonServer/FinalPythonShop/src/Item.py
# class  Item

class Item:
    # add stockerAdded: Stocker after, productID if we decide to put that in
    def __init__(self, name: str, product_id: int, type: str, quantity: int,
                 price: float, manufacturer: str, model: str):
        self._name = name
        self._product_id = product_id
        #self._stockerAdded = stockerAdded
        self._type = type
        self._quantity = quantity
        self._price = price
        self._manufacturer = manufacturer
        self._model = model

    @property
    def name(self):
        """Return the item's name"""
        return self._name
    
    @name.setter
    def name(self, name: str):
        self._name = name
        
    @property
    def product_id(self):
        """Return the product_id"""
        return self._product_id

    @property
    def type(self):
        """Return the type"""
        return self._type

    @property
    def quantity(self):
        """Return the quantity"""
        return self._quantity

    @quantity.setter
    def quantity(self, the_quantity):
        """Set the quantity"""
        if the_quantity >= 0:
            self._quantity = the_quantity
        else:
            raise ValueError('quantity must be a non-negative quantity')

    @property
    def price(self):
        """Return the price"""
        return self._price

    @price.setter
    def price(self, the_price):
        """Set the price"""
        if the_price >= 0:
            self._price = the_price
        else:
            raise ValueError('price must be a non-negative price')

    @property
    def manufacturer(self):
        """Return the manufacturer"""
        return self._manufacturer

    @property
    def model(self):
        """Return the model"""
        return self._model

    def __str__(self):
        """String representation of an Item object"""
        return "ITEM: [ItemName: {} | ID: {} | Type: {} | Quantity: {} | Price: {} | Manufacturer: {} | Model: {}]".format(
            self._name, self._product_id, self._type, self._quantity, self._price, self._manufacturer, self._model)

File: ClientDisplay/src/catalog/Item.java
package edu.cudenver.catalog;

// class Item

import java.io.Serializable;

public class Item implements Serializable {
    private int productID;                    // Haven't made const yet (static)
    //private static Stocker stockerAdded;    // Stocker class not defined yet
    private String type;
    private int quantity;
    private double price;                     // Changed to double, not updated on the UML
    private String manufacturer;
    private String model;

    // Constructor for Item     (STILL NEEDS STOCKER AFTER productID)
    public Item(int productID, String type, int quantity, double price, String manufacturer, String model){
        this.productID = productID;
        this.type = type;
        this.setQuantity(quantity);
        this.setPrice(price);
        this.manufacturer = manufacturer;
        this.model = model;
    }

    public int getProductID(){ return productID; }
    public void setProductID(int productID){ this.productID = productID; }

    public String getType() { return type; }
    public void setType(String type) {
        this.type = type;
    }

    public int getQuantity() { return quantity; }
    public void setQuantity(int quantity) throws IllegalArgumentException {
        if (quantity >= 0){
            this.quantity = quantity;
        } else {
            throw new IllegalArgumentException("Cannot have a negative quantity for an Item!");
        }
    }

    public double getPrice() { return price; }
    public void setPrice(double price) throws IllegalArgumentException {
        if (price >= 0){
            this.price = price;
        } else {
            throw new IllegalArgumentException("Cannot have a negative price for an Item!");
        }
    }

    public String getManufacturer() { return manufacturer; }
    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public String getModel() { return model; }
    public void setModel(String model) {
        this.model = model;
    }

    // Needs Stocker still after Stocker class added ToString()
    public String toString(){
        return String.format("ITEM: [ID: %d | Type: %s | Quantity: %d | Price: $%.2f | Manufacturer: %s | Model: %s]",
                this.getProductID(), this.getType(), this.getQuantity(), this.getPrice(), this.getManufacturer(), this.getModel());
    }

}   // end class Item